### ISSUE TITLE
Incorrect delete behavior on deleting whitespace

### ISSUE THREAD

**Issue by sachinrekhi** [quilljs/quill#833]
Generally delete and backspace do the right thing in Quill (ie. backspace deletes the content before the cursor and delete deletes the content after the cursor).

However I've noticed that delete does not work correctly when deleting an empty line with respect to cursor position. When you press delete on an empty line, the cursor position should remain on that line and the content on the subsequent line should come up. While the content from the subsequent line does come up, the cursor is now on the previous line.

As a sanity check, I compared my expected behavior to Google Docs and it does what I'm suggesting is the correct behavior.

**Steps for Reproduction**

1. Visit beta.quilljs.com
2. Clear editor
3. Type:
```
test

again
```
4. Place cursor on empty line between test and again
5. Hit delete

**Expected behavior**:
The empty line is deleted and the cursor is at the beginning of again

**Actual behavior**:
The empty line is delete but the cursor position is at the end of test

**Platforms**:
Chrome 51 on Mac 10.11

**Version**:
Quill 1.0 Beta 9

**Comment by benbro** [comment]
Unless I'm missing something, I think that the cursor should be at the end of test and quill do the right thing.
Textarea on browsers and Libreoffice act the same.

Cursor on empty line:
```
test
|
again
```

After backspace:
```
test|
again
```

**Comment by sachinrekhi** [comment]
@benbro keep in mind I'm distinguishing between DELETE and BACKSPACE here. What you are showing should certainly be the behavior for DELETE. But BACKSPACE typically has the behavior I'm describing.

Try it in this Github issues editor. You'll notice that with DELETE you get this:

```
testI
again
```

And with BACKSPACE you get this:

```
test
Iagain
```

Also holds true for Google Docs, Gmail compose window, etc.

**Comment by benbro** [comment]
For me it's the opposite.
After DELETE the cursor is on the second row.
After BACKSPACE the cursor is on the first row.

I'm on Windows 7.
Maybe it's a difference between Mac and Windows?

**Comment by sachinrekhi** [comment]
Oh you are right, I have the names wrong. What's confusing about a Mac is both of the keys actually say "delete" ;)

So the reality is that BACKSPACE works as expected, but DELETE is not functioning as expected in Quill.

**Comment by benbro** [comment]
I agree. After DELETE on Windows 7, the cursor should be on the second row but quill place the cursor on the first row in both DELETE and BACKSPACE.

**Comment by jhchen** [comment]
Thanks guys. This has been fixed in 1.0.0-beta.10.
